src/microbe_stage/PlayerHoverInfo.cs

```diff
@@ -29,7 +37,46 @@ public void Init(MicrobeCamera camera, CompoundCloudSystem cloudSystem)
 
     public override void _Process(float delta)
     {
-        HoveredCompounds = cloudSystem.GetAllAvailableAt(camera.CursorWorldPos);
+        // https://github.com/Revolutionary-Games/Thrive/issues/1976
+        if (delta <= 0)
+            return;
+
+        cloudSystem.GetAllAvailableAt(camera.CursorWorldPos, currentHoveredCompounds);
+
+        if (camera.CursorWorldPos != lastCursorWorldPos)
+        {
+            HoveredCompounds.Clear();
+            lastCursorWorldPos = camera.CursorWorldPos;
+        }
+
+        foreach (var compound in currentHoveredCompounds)
+        {
+            HoveredCompounds.TryGetValue(compound.Key, out float oldAmount);
+
+            // Delay removing of label to reduce flickering.
+            if (compound.Value == 0f && oldAmount > 0f)
+            {
+                compoundDelayTimer.TryGetValue(compound.Key, out float delayDelta);
```

Comment:
If the TryGetValue fails, then this assumes that delayDelta was initialized to 0, did I get that right?

Reply:
Yes, TryGetValue assigns default value on fail which is 0.0f.